NewsApp/app/src/main/java/com/example/newsapp/data/repos/NewsRepository.kt

```diff
@@ -0,0 +1,16 @@
+package com.example.newsapp.data.repos
+
+import com.example.newsapp.data.entities.NewsEntity
+import com.example.newsapp.data.network.NewsApi
+import com.example.newsapp.data.room.NewsDao
+import com.example.newsapp.utils.convertToEntity
+
+class NewsRepository(private val api: NewsApi, private val newsDao: NewsDao) : NewsRepo {
+
+    override suspend fun getNews(): NewsEntity {
```

Comment:
I think we should check if the database is empty first? We need to figure out the flow of pulling the data from either the network or the cache.

Reply:
this could be a business decision. for example, update the data every hour. or every time the user is online..etc.